diff --git a/CMakeLists.txt b/CMakeLists.txt
index 109394f..2e8979e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,13 @@ if(POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
 endif()
 
+if(POLICY CMP0063)
+  cmake_policy(SET CMP0063 NEW)
+endif()
+
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
+
 # Hint: This can be set to enable custom find_package
 # search paths, probably best to set it when configuring
 # on the command line to cmake instead of setting it
