Index: /wxWidgets/trunk/src/include/wx/platform.h
===================================================================
--- /wxWidgets/trunk/include/wx/platform.h	2013-05-04 09:12:20.542722088 +0200
+++ /wxWidgets/trunk/include/wx/platform.h	2013-05-04 09:12:10.318721872 +0200
@@ -738,7 +738,7 @@
  * Presently, only Windows and GTK+ support wxEVT_MENU_OPEN.
  */
 #ifndef wxUSE_IDLEMENUUPDATES
-#    if (defined(__WXMSW__) || defined(__WXGTK__) || defined(__WXOSX__)) && !defined(__WXUNIVERSAL__)
+#    if (defined(__WXMSW__) || defined(__WXOSX__)) && !defined(__WXUNIVERSAL__)
 #        define wxUSE_IDLEMENUUPDATES 0
 #    else
 #        define wxUSE_IDLEMENUUPDATES 1
