--- FreeImage 2/Makefile.osx	2015-03-08 19:04:16.000000000 -0600
+++ FreeImage/Makefile.osx	2016-11-07 18:18:49.000000000 -0700
@@ -6,28 +6,23 @@
 
 # General configuration variables:
-CC_PPC = gcc-4.0
-CC_I386 = gcc-4.0
-CC_X86_64 = gcc-4.0
-CPP_PPC = g++-4.0
-CPP_I386 = g++-4.0
-CPP_X86_64 = g++-4.0
-COMPILERFLAGS = -Os -fexceptions -fvisibility=hidden -DNO_LCMS
-COMPILERFLAGS_PPC = -arch ppc
+CC_PPC = clang
+CC_I386 = clang
+CC_X86_64 = clang
+CPP_PPC = clang++
+CPP_I386 = clang++
+CPP_X86_64 = clang++
+COMPILERFLAGS = -Os -fexceptions -fvisibility=hidden -mmacosx-version-min=10.9 -stdlib=libc++ -DNO_LCMS -D__ANSI__ -DDISABLE_PERF_MEASUREMENT
 COMPILERFLAGS_I386 = -arch i386
 COMPILERFLAGS_X86_64 = -arch x86_64
 COMPILERPPFLAGS = -Wno-ctor-dtor-privacy
 INCLUDE += 
-INCLUDE_PPC = -isysroot /Developer/SDKs/MacOSX10.5.sdk
-INCLUDE_I386 = -isysroot /Developer/SDKs/MacOSX10.5.sdk
-INCLUDE_X86_64 = -isysroot /Developer/SDKs/MacOSX10.6.sdk
-CFLAGS_PPC = $(COMPILERFLAGS) $(COMPILERFLAGS_PPC) $(INCLUDE) $(INCLUDE_PPC)
+INCLUDE_I386 = 
+INCLUDE_X86_64 = 
 CFLAGS_I386 = $(COMPILERFLAGS) $(COMPILERFLAGS_I386) $(INCLUDE) $(INCLUDE_I386)
 CFLAGS_X86_64 = $(COMPILERFLAGS) $(COMPILERFLAGS_X86_64) $(INCLUDE) $(INCLUDE_X86_64)
-CPPFLAGS_PPC = $(COMPILERPPFLAGS) $(CFLAGS_PPC)
 CPPFLAGS_I386 = $(COMPILERPPFLAGS) $(CFLAGS_I386)
 CPPFLAGS_X86_64 = $(COMPILERPPFLAGS) $(CFLAGS_X86_64)
-LIBRARIES_PPC = -Wl,-syslibroot /Developer/SDKs/MacOSX10.5.sdk
-LIBRARIES_I386 = -Wl,-syslibroot /Developer/SDKs/MacOSX10.5.sdk
-LIBRARIES_X86_64 = -Wl,-syslibroot /Developer/SDKs/MacOSX10.6.sdk
+LIBRARIES_I386 = 
+LIBRARIES_X86_64 = 
 LIBTOOL = libtool
 LIPO = lipo
@@ -39,9 +34,7 @@
 HEADER = Source/FreeImage.h
 
-.SUFFIXES: .o-ppc .o-i386 .o-x86_64
-MODULES_PPC = $(SRCS:.c=.o-ppc)
+.SUFFIXES: .o-i386 .o-x86_64
 MODULES_I386 = $(SRCS:.c=.o-i386)
 MODULES_X86_64 = $(SRCS:.c=.o-x86_64)
-MODULES_PPC := $(MODULES_PPC:.cpp=.o-ppc)
 MODULES_I386 := $(MODULES_I386:.cpp=.o-i386)
 MODULES_X86_64 := $(MODULES_X86_64:.cpp=.o-x86_64)
@@ -63,9 +56,6 @@
 FreeImage: $(STATICLIB) $(SHAREDLIB)
 
-$(STATICLIB): $(STATICLIB)-ppc $(STATICLIB)-i386 $(STATICLIB)-x86_64
-	$(LIPO) -create $(STATICLIB)-ppc $(STATICLIB)-i386 $(STATICLIB)-x86_64 -output $(STATICLIB)
-
-$(STATICLIB)-ppc: $(MODULES_PPC)
-	$(LIBTOOL) -arch_only ppc -o $@ $(MODULES_PPC)
+$(STATICLIB): $(STATICLIB)-i386 $(STATICLIB)-x86_64
+	$(LIPO) -create $(STATICLIB)-i386 $(STATICLIB)-x86_64 -output $(STATICLIB)
 
 $(STATICLIB)-i386: $(MODULES_I386)
@@ -75,9 +65,6 @@
 	$(LIBTOOL) -arch_only x86_64 -o $@ $(MODULES_X86_64)
 
-$(SHAREDLIB): $(SHAREDLIB)-ppc $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64
-	$(LIPO) -create $(SHAREDLIB)-ppc $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64 -output $(SHAREDLIB)
-
-$(SHAREDLIB)-ppc: $(MODULES_PPC)
-	$(CPP_PPC) -arch ppc -dynamiclib $(LIBRARIES_PPC) -o $@ $(MODULES_PPC)
+$(SHAREDLIB): $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64
+	$(LIPO) -create $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64 -output $(SHAREDLIB)
 
 $(SHAREDLIB)-i386: $(MODULES_I386)
@@ -87,7 +74,4 @@
 	$(CPP_X86_64) -arch x86_64 -dynamiclib $(LIBRARIES_X86_64) -o $@ $(MODULES_X86_64)
 
-.c.o-ppc:
-	$(CC_PPC) $(CFLAGS_PPC) -c $< -o $@
-
 .c.o-i386:
 	$(CC_I386) $(CFLAGS_I386) -c $< -o $@
@@ -96,7 +80,4 @@
 	$(CC_X86_64) $(CFLAGS_X86_64) -c $< -o $@
 
-.cpp.o-ppc:
-	$(CPP_PPC) $(CPPFLAGS_PPC) -c $< -o $@
-
 .cpp.o-i386:
 	$(CPP_I386) $(CPPFLAGS_I386) -c $< -o $@
@@ -113,3 +94,3 @@
 
 clean:
-	rm -f core Dist/*.* u2dtmp* $(MODULES_PPC) $(MODULES_I386) $(MODULES_X86_64) $(STATICLIB) $(STATICLIB)-ppc $(STATICLIB)-i386 $(SHAREDLIB) $(SHAREDLIB)-ppc $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64
+	rm -f core Dist/*.* u2dtmp* $(MODULES_PPC) $(MODULES_I386) $(MODULES_X86_64) $(STATICLIB) $(STATICLIB)-i386 $(SHAREDLIB) $(SHAREDLIB)-i386 $(SHAREDLIB)-x86_64
