Index: /wxWidgets/trunk/src/osx/cocoa/window.mm
===================================================================
--- /wxWidgets/trunk/src/osx/cocoa/window.mm (revision 71916)
+++ /wxWidgets/trunk/src/osx/cocoa/window.mm (revision 72108)
@@ -2085,6 +2085,10 @@
     {
         CFMutableArrayRef typesarray = CFArrayCreateMutable(kCFAllocatorDefault,0,&kCFTypeArrayCallBacks);
-        
-        [m_osxView registerForDraggedTypes:(NSArray*)typesarray];
+        dobj->AddSupportedTypes(typesarray);
+        NSView* targetView = m_osxView;
+        if ( [m_osxView isKindOfClass:[NSScrollView class] ] )
+            targetView = [(NSScrollView*) m_osxView documentView];
+
+        [targetView registerForDraggedTypes:(NSArray*)typesarray];
         CFRelease(typesarray);
     }
Index: /wxWidgets/trunk/src/osx/cocoa/listbox.mm
===================================================================
--- /wxWidgets/trunk/src/osx/cocoa/listbox.mm (revision 71808)
+++ /wxWidgets/trunk/src/osx/cocoa/listbox.mm (revision 72108)
@@ -556,6 +556,6 @@
 
     // temporary hook for dnd
-    [tableview registerForDraggedTypes:[NSArray arrayWithObjects:
-        NSStringPboardType, NSFilenamesPboardType, (NSString*) kPasteboardTypeFileURLPromise, NSTIFFPboardType, NSPICTPboardType, NSPDFPboardType, nil]];
+ //   [tableview registerForDraggedTypes:[NSArray arrayWithObjects:
+ //       NSStringPboardType, NSFilenamesPboardType, (NSString*) kPasteboardTypeFileURLPromise, NSTIFFPboardType, NSPICTPboardType, NSPDFPboardType, nil]];
 
     [ds setImplementation:c];
