Index: /wxWidgets/trunk/src/osx/cocoa/glcanvas.mm
===================================================================
--- /wxWidgets/trunk/src/osx/cocoa/glcanvas_old.mm	2013-04-26 13:24:07.000000000 +0200
+++ /wxWidgets/trunk/src/osx/cocoa/glcanvas.mm	2013-04-26 13:24:37.000000000 +0200
@@ -196,7 +196,7 @@
                     if ( (data[p++] = (NSOpenGLPixelFormatAttribute) attribList[arg++]) == true )
                     {
                         // don't use software fallback
-                        data[p++] = NSOpenGLPFANoRecovery;
+                        data[p++] = NSOpenGLPFAAccelerated;
                     }
                     break;
 
