--- glcanvas_orig.mm	2013-07-16 17:47:14.000000000 +0200
+++ glcanvas.mm	2013-09-25 20:02:54.000000000 +0200
@@ -89,6 +89,7 @@
         NSOpenGLPFAColorSize,(NSOpenGLPixelFormatAttribute)8,
         NSOpenGLPFAAlphaSize,(NSOpenGLPixelFormatAttribute)0,
         NSOpenGLPFADepthSize,(NSOpenGLPixelFormatAttribute)8,
+        NSOpenGLPFAAccelerated, // use hardware accelerated context
         (NSOpenGLPixelFormatAttribute)nil
     };
 
@@ -101,6 +102,7 @@
     {
         unsigned p = 0;
         data[p++] = NSOpenGLPFAMinimumPolicy; // make _SIZE tags behave more like GLX
+        data[p++] = NSOpenGLPFAAccelerated; // use hardware accelerated context
 
         for ( unsigned arg = 0; attribList[arg] !=0 && p < WXSIZEOF(data); )
         {
