--- src/osx/dnd_osx_old.cpp	2014-01-25 19:57:18.000000000 +0100
+++ src/osx/dnd_osx.cpp	2014-01-25 19:38:12.000000000 +0100
@@ -129,7 +129,7 @@
                     }
                     else
                     {
-                        char *d = new char[size];
+                        char *d = new char[size+1];
                         data->GetDataHere( format, (void*)d );
                         m_dataObject->SetData( format, size, d );
                         delete [] d;
