diff --git src/osx/textctrl_osx.cpp src/osx/textctrl_osx.cpp
index 1ef8afa420..452d4272cf 100644
--- src/osx/textctrl_osx.cpp
+++ src/osx/textctrl_osx.cpp
@@ -342,6 +342,13 @@ void wxTextCtrl::OnDropFiles(wxDropFilesEvent& event)
 
 void wxTextCtrl::OnKeyDown(wxKeyEvent& event)
 {
+    int key = event.GetKeyCode();
+    if ( key == WXK_RETURN || key == WXK_NUMPAD_ENTER ) {
+        // HACK: Forward these keys to the OnChar handler
+        OnChar(event);
+        return;
+    }
+
     if ( event.GetModifiers() == wxMOD_CONTROL )
     {
         switch( event.GetKeyCode() )
