diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d001c6..d302d71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ else()
   # set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
 
   if (MSVC)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W3 /Zi /permissive-")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /permissive-")
   else ()
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wshadow -Wformat=2 -Wall -Wno-overlength-strings -pedantic")
   endif ()
